collector/dfsr.go

```diff
@@ -0,0 +1,777 @@
+// +build windows
+
+package collector
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+const (
+	dfsrVolumeSubsystem     = "dfsr_volume"
+	dfsrConnectionSubsystem = "dfsr_connection"
+	dfsrFoldersSubsystem    = "dfsr_folder"
+)
+
+func init() {
+	registerCollector(dfsrConnectionSubsystem, NewDFSRConnectionCollector, "DFS Replication Service Connections")
+	registerCollector(dfsrFoldersSubsystem, NewDFSRConnectionCollector, "DFS Replication Service Folders")
+	registerCollector(dfsrVolumeSubsystem, NewDFSRConnectionCollector, "DFS Replication Service Volumes")
```

Comment:
I'm not sure we want three distinct collectors for this - can't we use the subcollector pattern here?

Reply:
Do we have a subcollector example? Are we thinking a separate namespace I.E. `dfsr.volume` rather than the current `dfsr_volume`?